http-server/src/test/clojure/xtdb/remote_test.clj

```diff
@@ -0,0 +1,217 @@
+(ns xtdb.remote-test
+  (:require [clojure.test :as t :refer [deftest]]
+            [clojure.string :as str]
+            [jsonista.core :as json]
+            [cognitect.transit :as transit]
+            [juxt.clojars-mirrors.hato.v0v8v2.hato.client :as http]
+            [xtdb.api :as xt]
+            [xtdb.client.impl :as xtc]
+            [xtdb.test-util :as tu :refer [*node*]]
+            [xtdb.jackson :as jackson]
+            [xtdb.serde :as serde])
+  (:import (java.io ByteArrayInputStream EOFException)))
+
+;; ONLY put stuff here where remote DIFFERS to in-memory
+
+(t/use-fixtures :each tu/with-mock-clock tu/with-http-client-node)
+
+(deftest normalisation-option
+  (t/is (thrown-with-msg? IllegalArgumentException
+                          #"Illegal argument: "
+                          (xt/q *node* '(from :docs [first-name last-name])
+                                {:key-fn identity}))
+        "remote can not serialize arbitrary fns"))
+
+
+(defn- http-url [endpoint] (str "http://localhost:" tu/*http-port* "/" endpoint))
+
+(defn- decode-transit* [^String s]
+  (let [rdr (transit/reader (ByteArrayInputStream. (.getBytes s)) :json {:handlers serde/transit-read-handlers})]
+    (loop [res []]
+      (let [value-read (try
+                         (transit/read rdr)
+                         (catch RuntimeException e
+                           (if (instance? EOFException (.getCause e))
+                             ::finished
+                             (throw e))))]
+        (if (= ::finished value-read)
+          res
+          (recur (conj res value-read)))))))
+
+(defn- decode-transit [^String s] (first (decode-transit* s)))
+
+(def ^:private default-object-mapper
```

Comment:
worth us making the `xtdb.jackson` one available so that we're testing the real version?